socorro/processor/signature_utilities.py

```diff
@@ -93,6 +93,63 @@ def __init__(self, config, quit_check_callback=None):
         self.fixup_comma = re.compile(r',(?! )')
 
     #--------------------------------------------------------------------------
+    def _is_exception(
+        self,
+        exception_list,
+        remaining_original_line,
+        line_up_to_current_position
+    ):
+        for an_exception in exception_list:
+            if remaining_original_line.startswith(an_exception):
+                return True
+            if line_up_to_current_position.endswith(an_exception):
+                return True
+        return False
+
+    #--------------------------------------------------------------------------
+    def _collapse(
+        self,
+        function,
+        open_string,
+        replacement_open_string,
+        close_string,
+        replacement_close_string,
+        exception_substring_list=(),  # list of exceptions that shouldn't collapse
+    ):
```

Comment:
Even though the code is pretty straight forward, now that I've studied it for a while, it's not clear what it does. The name reveals a lot which is a good start. 
How about this for a doc string. 
```
Return a new string where certain enclosed blocks can be replaced. For example
you can turn `somestring[stuff]` into `somestring[]` or `somestring{replacement}`.
```